src/uct/ib/base/ib_iface.c

```diff
@@ -695,18 +712,44 @@ uct_ib_iface_roce_is_reachable(const uct_ib_device_gid_info_t *local_gid_info,
                                             &local_gid_info->gid,
                                             &sa_local) != UCS_OK)) {
         uct_iface_fill_info_str_buf(
-               params, "Couldn't convert local RoCE address to socket address");
+               params, "couldn't convert local RoCE address to socket address");
         return 0;
     }
 
     if (uct_ib_device_roce_gid_to_sockaddr(remote_ib_addr_af,
                                            remote_ib_addr + 1,
                                            &sa_remote) != UCS_OK) {
         uct_iface_fill_info_str_buf(
-               params, "Couldn't convert remote RoCE address to socket address");
+               params, "couldn't convert remote RoCE address to socket address");
+        return 0;
+    }
+
+    status = uct_ib_device_get_roce_ndev_name(dev, port_num,
+                                              local_gid_info->gid_index,
+                                              ndev_name, sizeof(ndev_name));
+    if (status != UCS_OK) {
+        uct_iface_fill_info_str_buf(
+               params, "couldn't get network interface name");
+        return 0;
+    }
+
+    if (iface->config.reachability_mode == UCT_IB_REACHABILITY_MODE_ROUTE) {
+        if (ucs_netlink_rule_exists(ndev_name, (struct sockaddr *)&sa_remote)) {
+            return 1;
+        }
+
+        uct_iface_fill_info_str_buf(
+               params, "remote address %s is not routable",
+               ucs_sockaddr_str((struct sockaddr *)&sa_remote, remote_str, 128));
+
         return 0;
     }
 
+    /* check for zero-sized netmask */
```

Comment:
This comment repeats the code below it, without adding any info.
Please either make it more informative, e.g. "An 0-bit netmask implies any address is reachable" or remove it.